cdap-app-fabric/src/main/java/co/cask/cdap/config/PreferencesStore.java

```diff
@@ -153,4 +169,15 @@ private String getMultipartKey(String... parts) {
     }
     return Bytes.toString(result);
   }
+
+  private void validateProfileExistsAndStatus(
+    NamespaceId namespaceId, Map<String, String> propertyMap) throws BadRequestException, NotFoundException {
+    Optional<ProfileId> profile = SystemArguments.getProfileIdFromArgs(namespaceId, propertyMap);
+    ProfileId profileId = profile.orElse(null);
```

Comment:
the point of Optional is to avoid dealing with nulls. You can replace the null check with isPresent()

Reply:
fixed